internal/tracing/init.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/go-logr/stdr"
+	"go.opentelemetry.io/contrib/exporters/autoexport"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
+
+	"github.com/opentofu/opentofu/version"
+)
+
+/*
+BEWARE! This is not a committed external interface.
```

Comment:
Why not make it live in internal then?

Reply:
I'm blind, nevermind!

Reply:
This comment was originally from https://github.com/opentofu/opentofu/commit/503896af54d6eacd685873488494e6f14067a2de, in which I was intending it to mean that anything new that begins happening when you run with the `OTEL_TRACES_EXPORTER` environment variable is not protected by compatibility promises.

This new placement of the comment does perhaps leave it a little ambiguous which "this" it is talking about, and so perhaps we could clarify that by replacing "this" with something like "OpenTofu's tracing output". :man_shrugging: